# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/error-support-macros/src/lib.rs
# Commit: 2147c5c81edf
# Full Hash: 2147c5c81edf9cd56c19e7e92dfebfcd77329978
# Author: Ryan VanderMeulen <ryanvm@gmail.com>
# Date: 2023-03-01 17:52 -0500
# Regressor Bug: 1818349
# File Overlap Count: 24
# Overlapping Files: third_party/rust/sync15/src/client/coll_update.rs, third_party/rust/sync15/src/client/mod.rs, third_party/rust/sync15/.cargo-checksum.json, third_party/rust/sync15/src/client/sync.rs, Cargo.toml
# Description:
#   Backed out changesets b05343a5b533 and 57b2cda7e0db (bug 1818349) for causing bug 1819723.
# ==============================================================================

diff -r d59b76766f0d -r 2147c5c81edf third_party/rust/error-support-macros/src/lib.rs
--- a/third_party/rust/error-support-macros/src/lib.rs	Wed Mar 01 05:53:30 2023 +0000
+++ b/third_party/rust/error-support-macros/src/lib.rs	Wed Mar 01 17:52:01 2023 -0500
@@ -48,11 +48,9 @@
 ///    }
 /// }
 ///
-/// // The `handle_error` macro maps from the error supplied in the mandatory argument
-/// // (ie, `Error` in this example) to the error returned by the function (`ExternalError`
-/// // in this example)
-/// #[handle_error(Error)]
+/// #[handle_error]
 /// fn do_something() -> std::result::Result<String, ExternalError> {
+///    // The `handle_error` macro maps from `Error` to `ExternalError`
 ///    Err(Error{})
 /// }
 ///
@@ -74,13 +72,18 @@
     arguments: &syn::AttributeArgs,
 ) -> syn::Result<proc_macro2::TokenStream> {
     if let syn::Item::Fn(item_fn) = input {
-        let err_path = argument::validate(arguments)?;
+        argument::validate(arguments)?;
         let original_body = &item_fn.block;
 
         let mut new_fn = item_fn.clone();
         new_fn.block = parse_quote! {
             {
-                (|| -> ::std::result::Result<_, #err_path> {
+                // Note: the `Result` here is a smell
+                // because the macro is **assuming** a `Result` exists
+                // that reflects the return value of the block
+                // An improvement would include the error of the `original_block`
+                // as an attribute to the macro itself.
+                (|| -> Result<_> {
                     #original_body
                 })().map_err(::error_support::convert_log_report_error)
             }
@@ -92,7 +95,7 @@
     } else {
         Err(syn::Error::new(
             input.span(),
-            "#[handle_error(..)] can only be used on functions",
+            "#[handle_error] can only be used on functions",
         ))
     }
 }